test/TestCommandLineInvocationSameAsConfig.py

```diff
@@ -1,82 +1,104 @@
 
     def test_rulesdir_as_config(self):
-        self.assert_config_for("-r ../test/rules/", dict(rulesdir=["../test/rules/"]))
+        self.assert_config_equal(["-r", "test/fixtures/rules/"],
+                                 "test/fixtures/rulesdir.yml")
 
     def test_use_default_rules(self):
-        self.assert_config_for("-R -r ../test/rules/", dict(rulesdir=["../test/rules"],
-                                                            use_default_rules=True))
+        self.assert_config_equal(["-R", "-r", "test/fixtures/rules/"],
+                                 "test/fixtures/rulesdir-defaults.yml")
 
     def test_tags(self):
-        self.assert_config_for("-t skip_ansible_lint", dict(tags=["skip_ansible_lint"]))
+        self.assert_config_equal(["-t", "skip_ansible_lint"],
+                                 "test/fixtures/tags.yml")
 
     def test_verbosity(self):
-        self.assert_config_for("-v", dict(verbosity=1))
+        self.assert_config_equal(["-v"], "test/fixtures/verbosity.yml")
 
     def test_skip_list(self):
-        self.assert_config_for("-x bad_tag", dict(skip_list=["bad_tag"]))
+        self.assert_config_equal(["-x", "bad_tag"],
+                                 "test/fixtures/skip-tags.yml")
 
     def test_exclude(self):
-        self.assert_config_for("--exclude ../test/", dict(exclude_paths=["../test/"]))
+        self.assert_config_equal(["--exclude", "test/"],
+                                 "test/fixtures/exclude-paths.yml")
 
     def test_config_can_be_overridden(self):
-        no_override = self.run_ansible_lint(args="-t bad_tag")
-        overridden = self.run_ansible_lint(
-            args="-t bad_tag",
-            config=dict(tags=["skip_ansible_lint"]))
+        no_override, _ = cli.get_config(self.BASE_ARGUMENTS + ["-t", "bad_tag"])
 
-        self.assertEqual(no_override, overridden)
+        overridden, _ = cli.get_config(self.BASE_ARGUMENTS +
+                                       ["-t", "bad_tag",
+                                        "-c", "test/fixtures/tags.yml"])
+
+        self.assertEqual(no_override.tags + ["skip_ansible_lint"],
+                         overridden.tags)
 
     def test_different_config_file(self):
-        with open(".sandbox/subdir/ansible-config.yml", "w") as outfile:
-            yaml.dump(dict(verbosity=1), outfile, default_flow_style=False)
+        """Ensures an alternate config_file can be used."""
+        diff_config, _ = cli.get_config(self.BASE_ARGUMENTS +
+                                        ["-c", "test/fixtures/ansible-config.yml"])
+        no_config, _ = cli.get_config(self.BASE_ARGUMENTS + ["-v"])
+
+        self.assertEqual(diff_config.verbosity, no_config.verbosity)
+
+    def test_path_from_config_do_not_depend_on_cwd(self):  # Issue 572
+        current_dir = os.getcwd()
+
+        def restore_cwd():
+            os.chdir(current_dir)
+
+        self.addCleanup(restore_cwd)
```

Comment:
honestly, this'd look way cleaner if you just used pure pytest functions instead of unittest style... we're moving away from unittest so you might as well start new things w/o it